runtime/src/test/java/com/dylibso/chicory/runtime/WasmModuleTest.java

```diff
@@ -206,7 +206,13 @@ public void shouldSupportMemoryFactoryOverride() {
                 .withMemoryFactory(
                         limits -> {
                             memoryCreated.set(true);
-                            return new Memory(limits);
+                            return new ByteBufferMemory(limits) {
```

Comment:
I think it would be better to make `ByteBufferMemory` final

Reply:
What's the concern in letting people extend `ByteBufferMemory`?
Exposing too many implementation details?